src/engine/idTable/IdTable.h

```diff
@@ -224,20 +227,29 @@ class IdTable {
   // already set up with the correct number of columns and can be used directly.
   // If `NumColumns == 0` then the number of columns has to be specified via
   // `setNumColumns()`.
-  IdTable() requires(!isView && columnsAreAllocatable &&
-                     std::is_default_constructible_v<Allocator>)
+  CPP_template(typename = void)(
+      requires CPP_NOT(isView) CPP_and columnsAreAllocatable CPP_and
+          std::is_default_constructible_v<Allocator>) IdTable()
       : IdTable{NumColumns, Allocator{}} {};
-  explicit IdTable(Allocator allocator)
-      requires(!isView && columnsAreAllocatable)
+
+  CPP_template(typename = void)(
+      requires CPP_NOT(isView)
+          CPP_and columnsAreAllocatable) explicit IdTable(Allocator allocator)
       : IdTable{NumColumns, std::move(allocator)} {};
 
   // `IdTables` are expensive to copy, so we disable accidental copies as they
   // are most likely bugs. To explicitly copy an `IdTable`, the `clone()` member
   // function (see below) can be used.
-  IdTable(const IdTable&) requires(!isView) = delete;
-  IdTable& operator=(const IdTable&) requires(!isView) = delete;
+  CPP_template(typename = void)(requires(!isView))
+      IdTable(const IdTable&) = delete;
+
+  CPP_template(typename = void)(requires(!isView)) IdTable& operator=(
+      const IdTable&) requires(!isView)
+  = delete;
 
-  // Views are copyable, as they are cheap to copy.
+  //  Views are copyable, as they are cheap to copy.
+  // TODO: <ccoecontrol> what about these? template cannot be defaulted
+  // possible solution: implement defaulted ones by hand
```

Comment:
Thanks for pointing that out, In fact there seems to be no really simple + good C++-17 solution.
The manual implementation is rather error-prone, especially for the assignment.
I would suggest  putting this whole block in its original form behind a `#ifndef`
So
````
#ifdef QLEVER_CPP_17
  IdTable(const IdTable&) = default;
  IdTable& operator= (const IdTable&) = default;
#else
  // The original code with the requires
#endif
```
as soon as we compile in C++20 mode we will detect all the illegal copies.
And additionally write a comment tha4t there is no simple way to formulate this in C++17 .
(For completeness: We could add a separate layer of wrapper classes that only handles the viewness, but that would be overkill for now).